### PR TITLE
Improve gcc and include g++, clang and clang++

### PR DESCRIPTION
**Pull request by mcarton**
Examples for g++, clang and clang++ are just symlinks, as the syntax does not really change.

### CODE REVIEW

**Comment by srsudar** [comment]
Are the changes like of things like `main.c` to `*main.c*` so that they will be emphasized when rendered as markdown, or for something else?

**Comment by srsudar** [comment]
I'm going to hold off on merging this until I figure out #24, as links aren't currently being respected.

File: eg/examples/gcc.md [hunk 1]
```diff
@@ -1,30 +1,58 @@
...
+
+
+compile with size optimization enabled
+
+    gcc -Os main.c
+
+
+assemble *main.c* (default output is *main.s*)
+
+    gcc -S main.c
```
Comment (srsudar) [inline eg/examples/gcc.md]: Wow the `-S` flag is awesome! I didn't know about that one.

**Comment by mcarton** [comment]
Emphases was the idea indeed. It does not italicize in the terminal of course, but it does on Github, and it seemed more semantically correct to me.

**Pull request merged**